### ISSUE TITLE
createStubInstance - optional overrides

### ISSUE THREAD

**Issue by ifrost** [sinonjs/sinon#1857]
**Is your feature request related to a problem? Please describe.**
Stubbing multiple properties require to write the var name multiple times, for example:

```js
var stubbedObject = sinon.createStubInstance(MyClass);
stubbedObject.foo = sinon.stub().returns(2);
stubbedObject.bar = sinon.stub().callsFake(function() { return 1; });
stubbedObject.xyzzy = sinon.stub().returns('test');
```

**Describe the solution you'd like**
It'd be nice to have a shortcut syntax to provide optional overrides for createStubInstance, for example:

```js
var stubbedObject = sinon.createStubInstance(MyClass, {
  foo: sinon.stub().returns(2),
  bar: sinon.stub().callsFake(function() { return 1; }),
  xyzzy: sinon.stub().returns('test')
});
```

**Describe alternatives you've considered**
The same result could be achieved with "with":

```js
with (stubbedObject = sinon.createStubInstance(MyClass)) {
  foo = sinon.stub().returns(2);
  bar = sinon.stub().callsFake(function() { return 1; });
  xyzzy = sinon.stub().returns('test');
}
```

"with" expression, however, is marked as deprecated and probably it's not the best idea to use it.

Another alternative I thought about was to provide functions to give more flexibility:

```js
var stubbedObject = sinon.createStubInstance(MyClass, {
  foo: function(stub, property) { 
    stub[property] = sinon.stub().returns(2);
  }
});
```

The version above would be easier to reuse. For example, sometimes we use fluent interfaces:

```js
var stubbedObject = sinon.createStubInstance(MyClass);
stubbedObject.foo = sinon.stub().returnsThis();
stubbedObject.bar = sinon.stub().returnsThis();
stubbedObject.xyzzy = sinon.stub().returnsThis();
```

Could have even shorter syntax:

```js
var returnsThis = function(stub, propertyName) {
  stub[propertyName] = sinon.stub().returnsThis();
}

var stubbedObject = sinon.createStubInstance(MyClass, {
  foo: returnsThis,
  bar: returnsThis,
  xyzzy: returnsThis
});
```

**Comment by ifrost** [comment]
Hi, the PR for it is still open. Any chance of getting it merged?

**Comment by mroderick** [comment]
This was implemented in #1864 and published to NPM as `sinon@6.3.0`
